SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -27,7 +21,7 @@
 #include "SpatialCommandUtils.h"
 #include "SpatialGDKServicesConstants.h"
 #include "SpatialGDKServicesModule.h"
-#include "UObject/CoreNet.h"
+#include "Windows/MinWindows.h"
```

Comment:
So I'm a bit worried about this. Including this header leads to `winnt.h(603): [C4005] 'TEXT': macro redefinition` compilation warning which seems like it might break quite a bit of stuff?

Reply:
is this actually a problem?

Reply:
Mmmm, I think we ran into this before. We should find a solution to this but... I am not sure what that is...

Reply:
seems like all the stuff we need is in "WinUser.h" but when I only import that I get the error `  LocalDeploymentManager.cpp(506): [C3861] 'SendMessage': identifier not found` so I assume it has dependencies on some of the other files included in MinWindows :/

Reply:
This header won't exist on non windows platforms

Reply:
Is there no platform agnostic facilities that unreal offers that we can use? If not I suggest sticking this in a platform layer of our own, I'd prefer it if we silo'd platform specific code if possible.

Reply:
Ideally we would use some unreal library to allow us to work cross platform. I believe last time this was investigated we did find some promising unreal code but no working solution.

Reply:
How about: `void SpatialCommandUtils::TryKillProcessWithName(const FString& ProcessName)`

Reply:
I believe that terminates the process rather than sending a close signal? I could be wrong though...?

Reply:
hat tip Dmitrii

Reply:
Thinking about it a bit more maybe what you want is `FGenericWindow::Destroy()`

Reply:
How about leveraging: `FPlatformProcess::CloseProc`

Reply:
seems like closeproc isn't what we want either, looks like it was discussed in matt's pr https://github.com/spatialos/UnrealGDK/pull/2899

Reply:
though that might have changed as i think the reason it ended up being a net message was because there wasn't process signal support. It doesn't really sound like it requests for the program to close though, from the docs https://docs.microsoft.com/en-us/windows/win32/api/handleapi/nf-handleapi-closehandle
@MatthewSandfordImprobable any thoughts, reckon this might work?

Reply:
> Some success here but after more testing I am a little worried CloseProc isn't doing what we want it to. So calling SpatialCommandUtils::TryKillProcessWithName after CloseProc will actually terminate it. Brilliant. I am still not super convinced that the waiting logic is now valid. Also still concerned CloseProc does not actually do what we want it to do.

Reply:
might be worth considering.

Reply:
We might be able to look at the runtime logs and see if it prints out that it received an exit signal? This might help working out if we are closing or terminating.

Reply:
I did some more digging (with miron)

So it seems like `TerminateProc` on windows isn't what we want, as on windows it calls the windows API `TerminateProcess`, which immediately kills the process. However, on unix, TerminateProc sends a SIGTERM signal which I believe is what we want (but don't really have a way to test at the moment). So I've refactored the methods to SpatialCommandUtils.h and are using WM_CLOSE on windows, while using terminateproc on unix

lmk what you think